### PR TITLE
Update RN to 54

### PR DESCRIPTION
**Pull request by chrisnojima**
and update some deps

1. turned off video on react-native-image-picker as we don't handle it well on the reading side and we'd have to change our permissions i think
2. changed some imports. metro has a bug currently where if you import a file through different contexts (like blah.js and blah.native.js) it'll puke (https://github.com/facebook/metro/commit/233b00e8366b4429bee4104502339ccb7e7bcb91) . they'll fix this but easy to fix our side too
3. Updating / trying mainline react-native-contacts

react-native-camera had a big change and has a ton of stuff we don't want (face detection) and the install process is more complicated. i started doing this but it was taking too much time

### CODE REVIEW

**Comment by chrisnojima** [comment]
- [ ] android seems to ask then show nothign, but if you come back it works

**Comment by chrisnojima** [comment]
@buoyad maybe you can see about the android persmissions thing. the first time you get a blank contacts list even if you say yes, but then if you come back it'll be there

**Comment by chrisnojima** [comment]
@keybase/react-hackers this is ready

File: shared/react-native/android/app/src/main/java/io/keybase/ossifrage/MainActivity.java [hunk 1]
```diff
@@ -95,7 +95,7 @@ public void onRequestPermissionsResult(int requestCode, String[] permissions, in
         if (listener != null) {
             listener.onRequestPermissionsResult(requestCode, permissions, grantResults);
         }
-        if (permissions[0].equals("android.permission.READ_CONTACTS")) {
+        if (permissions.length > 1 && permissions[0].equals("android.permission.READ_CONTACTS")) {
```
Comment (akalin-keybase) [inline shared/react-native/android/app/src/main/java/io/keybase/ossifrage/MainActivity.java]: do you mean `> 0`?

Comment (buoyad) [reply shared/react-native/android/app/src/main/java/io/keybase/ossifrage/MainActivity.java]: yup!

Comment (buoyad) [reply shared/react-native/android/app/src/main/java/io/keybase/ossifrage/MainActivity.java]: fixed!

**Pull request merged**